pkg/dns/health.go

```diff
@@ -60,3 +61,62 @@ func (*FakeHealthCheckReconciler) Delete(ctx context.Context, _ *v1alpha1.Endpoi
 }
 
 var _ HealthCheckReconciler = &FakeHealthCheckReconciler{}
+
+type CachedHealthCheckReconciler struct {
+	reconciler HealthCheckReconciler
+	provider   Provider
+
+	cache map[string]HealthCheckSpec
```

Comment:
do we need to worry about concurrent access with this cache?

Reply:
It would probably be wiser to make it thread-safe but as the health check reconciler is only being used by the DNSPolicy controller it shouldn't be a problem

Reply:
Yes but I would imagine each object event is handled in its own routine to avoid blocking. So multiple go routines will access this shared map. Granted they may not clash but probably safer to protect access to it anyway

Reply:
actually I am wrong by default there is only 1 worker. However this can be tweaked with config

Reply:
so I think a threadsafe map makes sense as the default

Reply:
Thanks @maleck13 I just pushed a commit with the change using sync.Map as the cache